src/core/map.cpp

```diff
@@ -310,6 +310,12 @@ void Map::setDimensions(int newWidth, int newHeight, bool setNewBlockdata) {
         setNewDimensionsBlockdata(newWidth, newHeight);
     }
 
+    int oldWidth = layout->width.toInt();
+    int oldHeight = layout->height.toInt();
+    if (oldWidth != newWidth || oldHeight != newHeight) {
+        Scripting::cb_MapResized(oldWidth, oldHeight, newWidth, newHeight);
```

Comment:
Should this occur after the actual layout dimensions are changed?  Seems like it's happening too early here, and can mislead the script.